File: ncEfd/Persistencia/pReg0150.vb
Imports System.Text
Imports ncComum.nsAcessoBD
Imports ncComum.nsFuncoes.cFuncoes
Imports ncComum.nsExcecao

Namespace nsEfd

  Public Class pReg0150

    Public Function Consultar(ByVal dataInicio As Date, ByVal dataFim As Date) As Colecao0150

      Dim retorno As Colecao0150 = Nothing
      Dim acessoBanco As cAcessoBD = New cAcessoBD
      Dim ds As DataSet
      Dim dt As DataTable
      Dim comando As StringBuilder = New StringBuilder()

      comando.Append(" SELECT ")
      comando.Append(" f.cid, f.nome, f.cnpj, f.inscricaoEstadual, m.codigo_ibge, ")
      comando.Append(" f.logradouro, f.numero, f.complemento, f.bairro ")
      comando.Append(" FROM fornecedores f ")
      comando.Append(" INNER JOIN notafiscalfornecedor nff ON nff.fornecedor_cid = f.cid ")
      comando.Append(" INNER JOIN municipios m ON m.cid = f.cidade_cid ")
      comando.Append(" WHERE nff.dataEmissao BETWEEN '")
      comando.Append(FormatarDataUniversal(FormatarDataBarras(dataInicio)))
      comando.Append(" 00:00:00.000' AND '")
      comando.Append(FormatarDataUniversal(FormatarDataBarras(dataFim)))
      comando.Append(" 23:59:59.999' ")
      comando.Append(" ORDER BY f.cid ")

      ds = acessoBanco.ExecutarDS(comando.ToString())

      If Not ds Is Nothing Then
        If ds.Tables.Count > 0 Then
          dt = ds.Tables(0)

          If dt.Rows.Count > 0 Then
            retorno = New Colecao0150()

            For Each linha As DataRow In dt.Rows
              Dim _dReg0150 As dReg0150 = New dReg0150()

              _dReg0150.cod_part = RetornarTexto(linha.Item("cid"))
              _dReg0150.nome = RetornarTexto(linha.Item("nome"))
              _dReg0150.cnpj = RetornarTexto(linha.Item("cnpj"))
              _dReg0150.ie = RetornarTexto(linha.Item("inscricaoEstadual"))
              _dReg0150.cod_mun = RetornarTexto(linha.Item("codigo_ibge"))
              _dReg0150.ende = RetornarTexto(linha.Item("logradouro"))
              _dReg0150.num = RetornarTexto(linha.Item("numero"))
              _dReg0150.compl = RetornarTexto(linha.Item("complemento"))
              _dReg0150.bairro = RetornarTexto(linha.Item("bairro"))

              retorno.Add(_dReg0150)
            Next
          End If
        End If
      End If

      Consultar = retorno

    End Function

  End Class

End Namespace

File: nNefd/Persistencia/pReg0150.cs
using System;
using System.Data;
using System.Text;
using ncNComum.nsAcessoBD;
using nsEfd;


namespace nNefd.Persistencia
{
    public class pReg0150
    {
        public Colecao0150 Consultar(DateTime dataInicio, DateTime dataFim)
        {
            Colecao0150 retorno = null;
            var acessoBanco = new cAcessoBD();
            var comando = new StringBuilder();

            comando.Append(" SELECT ");
            comando.Append(" f.cid, f.nome, f.cnpj, f.inscricaoEstadual, m.codigo_ibge, ");
            comando.Append(" f.logradouro, f.numero, f.complemento, f.bairro ");
            comando.Append(" FROM fornecedores f ");
            comando.Append(" INNER JOIN notafiscalfornecedor nff ON nff.fornecedor_cid = f.cid ");
            comando.Append(" INNER JOIN municipios m ON m.cid = f.cidade_cid ");
            comando.Append(" WHERE nff.dataEmissao BETWEEN '");
            comando.Append(ncNComum.cFuncoes.FormatarDataUniversal(ncNComum.cFuncoes.FormatarDataBarras(dataInicio.ToString())));
            comando.Append(" 00:00:00.000' AND '");
            comando.Append(ncNComum.cFuncoes.FormatarDataUniversal(ncNComum.cFuncoes.FormatarDataBarras(dataFim.ToString())));
            comando.Append(" 23:59:59.999' ");
            comando.Append(" ORDER BY f.cid ");

            DataSet ds = acessoBanco.ExecutarDS(comando.ToString());

            if (ds != null && ds.Tables.Count > 0)
            {
                DataTable dt = ds.Tables[0];
                if (dt.Rows.Count > 0)
                {
                    retorno = new Colecao0150();
                    foreach (DataRow linha in dt.Rows)
                    {
                        var item = new dReg0150();
                        item.cod_part = ncNComum.cFuncoes.RetornarTexto(linha["cid"]);
                        item.nome = ncNComum.cFuncoes.RetornarTexto(linha["nome"]);
                        item.cnpj = ncNComum.cFuncoes.RetornarTexto(linha["cnpj"]);
                        item.ie = ncNComum.cFuncoes.RetornarTexto(linha["inscricaoEstadual"]);
                        item.cod_mun = ncNComum.cFuncoes.RetornarTexto(linha["codigo_ibge"]);
                        item.ende = ncNComum.cFuncoes.RetornarTexto(linha["logradouro"]);
                        item.num = ncNComum.cFuncoes.RetornarTexto(linha["numero"]);
                        item.compl = ncNComum.cFuncoes.RetornarTexto(linha["complemento"]);
                        item.bairro = ncNComum.cFuncoes.RetornarTexto(linha["bairro"]);
                        retorno.Add(item);
                    }
                }
            }

            return retorno;
        }
    }
}
